crates/turbopath/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+mod absolute_system_path_buf;
+mod anchored_system_path_buf;
+mod relative_system_path_buf;
+mod relative_unix_path_buf;
+
+use std::path::{Path, PathBuf};
+
+pub use absolute_system_path_buf::AbsoluteSystemPathBuf;
+pub use anchored_system_path_buf::AnchoredSystemPathBuf;
+use path_slash::{PathBufExt, PathExt};
+pub use relative_system_path_buf::RelativeSystemPathBuf;
+pub use relative_unix_path_buf::RelativeUnixPathBuf;
+use thiserror::Error;
+
+// Custom error type for path validation errors
+#[derive(Debug, Error)]
+pub enum PathValidationError {
+    #[error("Path is non-UTF-8")]
+    InvalidUnicode,
+    #[error("Path is not absolute: {0}")]
+    NotAbsolute(PathBuf),
+    #[error("Path is not relative: {0}")]
+    NotRelative(PathBuf),
+    #[error("Path {0} is not parent of {1}")]
+    NotParent(String, String),
+}
+
+trait IntoSystem {
+    fn into_system(self) -> Result<PathBuf, PathValidationError>;
+}
+
+trait IntoUnix {
+    fn into_unix(self) -> Result<PathBuf, PathValidationError>;
+}
+
+impl IntoSystem for &Path {
+    fn into_system(self) -> Result<PathBuf, PathValidationError> {
+        let path_str = self.to_str().ok_or(PathValidationError::InvalidUnicode)?;
+
+        Ok(PathBuf::from_slash(path_str))
+    }
+}
+
+impl IntoUnix for &Path {
+    /// NOTE: `into_unix` *only* converts Windows paths to Unix paths *on* a
```

Comment:
FWIW, `turbo` doesn't really have the notion of a "windows path". There's "system path" which is a path native to whatever OS you're currently running on, and "Unix path" which is treated as a portable format (and as such should always be relative).

If you're on a non-windows system, you shouldn't really have to worry about windows paths. They shouldn't occur in a situation where you are expected to manipulate them, and if they do, it's likely a bug.